source/expressions.tex

```diff
@@ -5540,10 +5540,12 @@
 \pnum
 If the keyword \keyword{delete} in a \grammarterm{delete-expression}
 is not preceded by the unary \tcode{::} operator and the type of the operand is
-a pointer to a (possibly cv-qualified) class type \tcode{T}:
+a pointer to a (possibly cv-qualified) class type \tcode{T}
+or (possibly multidimensional) array thereof:
 \begin{itemize}
 \item
-If \tcode{T} has a virtual destructor,
+If the operand is a pointer to \cv{} \tcode{T} and
```

Comment:
````cpp
auto ptr = new A[2];
delete[] ptr;
`````
`ptr` is a pointer to the initial element of the array. Hence, we could say it is a pointer to `T`. However, I think the lookup deallocation function in this case is still `operator delete[]`.

Reply:
Fixed